src/media/local-stream.ts

```diff
@@ -60,6 +79,50 @@ abstract class _LocalStream extends Stream {
       this[StreamEventNames.Muted].emit(isMuted);
     }
   }
+
+  /**
+   * Adds an effect to a local stream.
+   *
+   * @param name - The name of the effect.
+   * @param effect - The effect to add.
+   */
+  async addEffect(name: string, effect: TrackEffect): Promise<void> {
+    const outputTrack = await effect.load(this.outputStream.getTracks()[0]);
+    this.effects.push({ name, effect });
+    replaceTrack(this.outputStream, outputTrack);
+
+    // When the effect's track is updated, update the next effect or output stream.
+    effect.on(EffectEvent.TrackUpdated, (track: MediaStreamTrack) => {
+      const effectIndex = this.effects.findIndex((i) => i.name === name);
+      if (effectIndex === this.effects.length - 1) {
+        replaceTrack(this.outputStream, track);
+      } else {
+        this.effects[effectIndex + 1]?.effect.replaceInputTrack(track);
+      }
+    });
```

Comment:
Do we need to do the 'pendingEffect' scheme that was recently added in webrtc-core main?

Reply:
Added.